test/e2e/e2e_test.go

```diff
@@ -99,85 +99,121 @@ func RcByNameContainer(name string, replicas int32, labels map[string]string, gr
+
+	nodes, err := nodeutil.ReadyNodes(ctx, clientset, nodeInformer, "", nil)
 	if err != nil {
 		klog.Fatalf("%v", err)
 	}
 
-	lowNodeUtilizationStrategy := deschedulerapi.DeschedulerStrategy{
-		Enabled: true,
-		Params: &deschedulerapi.StrategyParameters{
-			NodeResourceUtilizationThresholds: &deschedulerapi.NodeResourceUtilizationThresholds{
-				Thresholds:       thresholds,
-				TargetThresholds: targetThresholds,
+	strategies.LowNodeUtilization(
+		ctx,
+		clientset,
+		deschedulerapi.DeschedulerStrategy{
+			Enabled: true,
+			Params: &deschedulerapi.StrategyParameters{
+				NodeResourceUtilizationThresholds: &deschedulerapi.NodeResourceUtilizationThresholds{
+					Thresholds: deschedulerapi.ResourceThresholds{
+						v1.ResourceMemory: 20,
+						v1.ResourcePods:   20,
+						v1.ResourceCPU:    85,
+					},
+					TargetThresholds: deschedulerapi.ResourceThresholds{
+						v1.ResourceMemory: 20,
+						v1.ResourcePods:   20,
+						v1.ResourceCPU:    90,
+					},
+				},
 			},
 		},
-	}
-
-	podEvictor := evictions.NewPodEvictor(
-		clientset,
-		evictionPolicyGroupVersion,
-		false,
-		0,
 		nodes,
-		false,
+		evictions.NewPodEvictor(
+			clientset,
+			evictionPolicyGroupVersion,
+			false,
+			0,
+			nodes,
+			false,
+		),
 	)
 
-	strategies.LowNodeUtilization(ctx, clientset, lowNodeUtilizationStrategy, nodes, podEvictor)
 	time.Sleep(10 * time.Second)
 }
 
-func TestE2E(t *testing.T) {
-	// If we have reached here, it means cluster would have been already setup and the kubeconfig file should
-	// be in /tmp directory as admin.conf.
+func TestLowNodeUtilization(t *testing.T) {
 	ctx := context.Background()
+
 	clientSet, err := client.CreateClient(os.Getenv("KUBECONFIG"))
 	if err != nil {
 		t.Errorf("Error during client creation with %v", err)
 	}
 
+	stopChannel := make(chan struct{}, 0)
+
+	sharedInformerFactory := informers.NewSharedInformerFactory(clientSet, 0)
+	sharedInformerFactory.Start(stopChannel)
+	sharedInformerFactory.WaitForCacheSync(stopChannel)
+	defer close(stopChannel)
+
+	nodeInformer := sharedInformerFactory.Core().V1().Nodes()
+
 	nodeList, err := clientSet.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
 	if err != nil {
 		t.Errorf("Error listing node with %v", err)
 	}
+
 	var nodes []*v1.Node
 	for i := range nodeList.Items {
 		node := nodeList.Items[i]
 		nodes = append(nodes, &node)
 	}
-	sharedInformerFactory := informers.NewSharedInformerFactory(clientSet, 0)
-	nodeInformer := sharedInformerFactory.Core().V1().Nodes()
+
+	rc := RcByNameContainer("test-rc-node-utilization", int32(15), map[string]string{"test": "node-utilization"}, nil)
+	if _, err := clientSet.CoreV1().ReplicationControllers(rc.Namespace).Create(ctx, rc, metav1.CreateOptions{}); err != nil {
+		t.Errorf("Error creating deployment %v", err)
+	}
+
+	evictPods(ctx, t, clientSet, nodeInformer, nodes, rc)
+	deleteRC(ctx, t, clientSet, rc)
+
+	// wait until rc is deleted
+	time.Sleep(5 * time.Second)
```

Comment:
Instead of time.Sleep maybe this would be better with a wait.Poll that actually checks for when the RC is deleted?